### PR TITLE
examples/video_player: Add video player

### PR DESCRIPTION
**Pull request by ahomescu**
Port the video player example from Phase 1
written in Rust to C.

### CODE REVIEW

**Review by endbr64 — APPROVED** [review]
Not an expert on FFMPEG/SDL, but LGTM!

File: cmake/define-test.cmake [hunk 1]
```diff
@@ -140,6 +140,7 @@ endfunction()
 
 # NEEDS_LD_WRAP - If present pass -Wl,@$LD_ARGS_FILE to the wrapped
 #       library. The name of the file is generated from the target name and pkey.
+# NOT_IN_CHECK_IA2 - If present, skip adding this target to check-ia2.
```
[the remark this answers is not in the document]
Comment (ahomescu) [reply cmake/define-test.cmake:143]: There are several differences:
* It takes a file name as its command line argument, none of the other tests do that. You can't run the executable by itself. I guess we could add a reference video file, but is it worth it?
* It's a GUI program that opens an X window, which is pointless in CI.

Tl;dr: I don't think we should run it in CI. However, we do need to build it, and the build system infrastructure for tests is also useful for the video player (and maybe other future examples).

File: cmake/define-test.cmake [hunk 2]
```diff
@@ -20,10 +11,15 @@ set(PADDING_LINKER_SCRIPT ${libia2_BINARY_DIR}/padding.ld)
 #       specified, this must be relative to the source directory for the test.
 #       Defaults to include/. If headers are also used for main executable, this
 #       must not be specified.
+# UNWRAPPED_INCLUDE_DIRS (optional) - extra directories with unwrapped headers.
```
[the remark this answers is not in the document]
Comment (ahomescu) [reply cmake/define-test.cmake:14]: They run in the same compartment as whatever calls them, no switching. Concretely for the video player:
* SDL2 runs in the same compartment as the main program
* The FFmpeg libraries run mostly in the same compartment as the video_decoder library.

I kinda want to call them "system libraries" instead, but they don't have to be that necessarily.

File: cmake/define-test.cmake (hunk 1 above)
[the remark this answers is not in the document]
Comment (rinon) [reply cmake/define-test.cmake:143]: We technically can add the binary target to the check-ia2 target, and that won't automatically run anything. I'd prefer we do this in a separate rule that isn't called "define-test." So I would suggest we just skip this extra param, add it to the target, and do something for non-test binaries later.
